pkgs/development/tools/gomacro/default.nix

```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gomacro";
+  rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
+  version = "20210131-${lib.strings.substring 0 7 rev}";
```

Comment:
I'm not 100% sure I did this correctly. I verified this works, but would it be preferred to do something like
```nix
let
rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
in
buildGoModule rec {
    pname = "gomacro";
    version = "20210131-${lib.strings.substring 0 7 rev}";
```
...etc

@fabaff @SuperSandro2000

Reply:
You can leave it like this. let in is not strictly required here.